### PR TITLE
[libc++] Add missing conditionals for feature-test macros

### PR DESCRIPTION
**Pull request by ldionne**
We noticed that some feature-test macros were not conditional on configuration flags like _LIBCPP_HAS_NO_FILESYSTEM. As a result, code attempting to use FTMs would not work as intended.

This patch adds conditionals for a few feature-test macros, but more issues may exist.

rdar://122020466

### CODE REVIEW

File: libcxx/utils/generate_feature_test_macro_components.py [hunk 1]
```diff
@@ -875,11 +877,14 @@ def add_version_header(tc):
             "name": "__cpp_lib_print",
             "values": {"c++23": 202207},
             "headers": ["ostream", "print"],
+            # TODO: Are we missing something here?
```
Comment (ldionne) [inline libcxx/utils/generate_feature_test_macro_components.py:880]: @mordante

Comment (mordante) [reply libcxx/utils/generate_feature_test_macro_components.py:880]: This matches https://isocpp.org/std/standing-documents/sd-6-sg10-feature-test-recommendations#__cpp_lib_print 
The other paper affecting print https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2539r4.html does not change a FTM. What do you think we might miss?

**Review by mordante — APPROVED** [review]
From a manual review this LGTM. I wonder whether we can test this better in the CI.

File: libcxx/utils/generate_feature_test_macro_components.py (hunk 1 above)
Comment (ldionne) [reply libcxx/utils/generate_feature_test_macro_components.py:880]: I was wondering whether we were missing e.g. `_LIBCPP_HAS_NO_LOCALIZATION` or something like that.

**Comment by ldionne** [comment]
> From a manual review this LGTM. I wonder whether we can test this better in the CI.

I'm not certain I follow what you mean. Do you mean that since we generate those tests automatically, if something's wrong in how we generate the tests, then we might not actually be testing the right thing?

**Pull request merged**